kqueen/engines/heat.py

```diff
@@ -0,0 +1,178 @@
+                    "required": True,
+                },
+            },
+            "username": {
+                "type": "text",
+                "label": "Username",
+                "order": 20,
+                "validators": {
+                    "required": True,
+                },
+            },
+            "password": {
+                "type": "password",
+                "label": "Password",
+                "order": 30,
+                "validators": {
+                    "required": True,
+                },
+            },
+            "domain_name": {
+                "type": "text",
+                "label": "Domain name",
+                "order": 40,
+                "validators": {
+                    "required": True,
+                },
+            },
+            "project_id": {
+                "type": "text",
+                "label": "Project ID",
+                "order": 50,
+                "validators": {
+                    "required": True,
+                },
+            },
+        }
+    }
+
+    def __init__(self, cluster, *args, **kwargs):
+        super().__init__(cluster, *args, **kwargs)
+        self.kwargs = kwargs
+        self.stack_name = "Kqueen_stack_" + cluster.id
+        self.heat = Heat(
+            auth_url=kwargs["auth_url"],
+            project_id=kwargs["project_id"],
+            username=kwargs["username"],
+            compute_api_version='3',
+            domain_name=kwargs["domain_name"],
+            identity_interface='public',
+            password=kwargs["password"],
+        )
+
+    def provision(self, *args, **kwargs):
+        return self.heat.create(name=self.stack_name,
+                template_name=self.kwargs["template"],
+                ssh_key_name="et",
+                server_image="centos-atomic-7")
+
+    def deprovision(self):
+        return self.heat.delete(self.stack_name)
+
+    def resize(self, *args, **kwargs):
+        pass
+    
+    def get_kubeconfig(self):
+        pass
+
+    def cluster_get(self):
+        stack = self.heat.get(self.stack_name)
+        if stack is None:
+            state = config.get('CLUSTER_UNKNOWN_STATE')
+        else:
+            state = STATE_MAP.get(stack["status"], config.get('CLUSTER_UNKNOWN_STATE'))
+        return {
+                'key': self.stack_name,   # (str) this record should be cached under this key if you choose to cache
+                'name': self.stack_name,  # (str) name of the cluster in its respective backend
+                'id': self.cluster.id,    # (str or UUID) id of `kqueen.models.Cluster` object in KQueen database
+                'state': state, # (str) state of cluster on backend represented by app.config['CLUSTER_[FOO]_STATE']
+                'metadata': {}, # any keys specific for the Provisioner implementation
+        }
+
+    def cluster_list(self):
+        return []
+
+    @classmethod
+    def engine_status(cls, **kwargs):
+        return config.get('PROVISIONER_OK_STATE')
```

Comment:
its just stub?

Reply:
in this method we need to ping cloud and try to auth with defined creds and return OK if success both, UNKNOWN if ping success only, and error otherwise

Reply:
see gke/jenkins methods

Reply:
this can be implemented in a separate commit, just set the todo please.